### PR TITLE
Reuse new implementation when is possible for filtering

### PR DESCRIPTION
**Pull request by MacondoExpress**
# Description

This PR migrates the old translation for filtering defined in `create-where-and-param` in favor of the new approach introduced with the QueryAST. 

## Complexity

Complexity: Med

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [x] Documentation has been updated
- [x] TCK tests have been updated
- [x] Integration tests have been updated
- [x] Example applications have been updated
- [x] New files have copyright header
- [x] CLA (https://neo4j.com/developer/cla/) has been signed

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the Neo4j GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-4213.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

File: packages/graphql/src/translate/authorization/where/create-authorization-jwt-payload-predicate.ts [hunk 1]
```diff
@@ -64,7 +62,7 @@ export function createJwtPayloadWherePredicate({
     return Cypher.and(...predicates);
 }
 
-function createNestedPredicate({
+function createNestedPredicateLegacy({
```
Comment (angrykoala) [inline packages/graphql/src/translate/authorization/where/create-authorization-jwt-payload-predicate.ts:65]: Additionally, we could mark this as `@deprecated`

```suggestion
/**
 * @deprecated
 **/
function createNestedPredicateLegacy({
```

File: packages/graphql/src/utils/join-predicates.ts [hunk 2]
```diff
@@ -1,43 +0,0 @@
...
-    if (filteredPredicates.length === 1) {
-        return filteredPredicates[0] as string;
-    }
-
-    return `(${
-        joinType === "AND"
-            ? filteredPredicates.join(" AND ")
-            : filteredPredicates.map((predicate) => `(${predicate})`).join(" OR ")
-    })`;
-}
```
Comment (angrykoala) [inline packages/graphql/src/utils/join-predicates.ts:43]: Ooh, we are getting rid of Cypher-Builder v0.0.1

**Review by angrykoala — APPROVED** [review]

File: packages/graphql/src/utils/join-predicates.ts (hunk 2 above)
Comment (MacondoExpress) [reply packages/graphql/src/utils/join-predicates.ts:43]: We do indeed!

**Pull request merged**